Check for Zoom Meetings, --check-activity

amadotejada opened the pull request:
If active Zoom meetings are detected, the script will exit. No user dialogs are shown. This handles scenarios when we run the script remotely via MDM tools in alternative to the Self-Service method, I think it's ideal to only display dialog prompts to users when they are not in Zoom meetings, Apple Silicon, password prompt, etc. Other video conference solutions could also be added.

grahampugh commented:
Hi, thanks for the PR, it's an interesting idea. A couple of points:

- functions are in their own part of the script (and mostly in alphabetical order), not inline with the main script body. 
- I would not expect a different function for each possible variety of video communication app, so I think the parameter label should be more generic. Something like --check-active-meetings or --check-activity (could include PowerPoint etc if that's possible).

amadotejada commented:
Hello @grahampugh, 

Thanks for your points. I've done the following.

- renamed to `--check-activity`
- moved function and in alphabetical order

grahampugh commented:
Thanks, there's one error but I'll merge it in and make the change myself.
